{
  "comments": [
    {
      "unresolved": false,
      "key": {
        "uuid": "33d1117c_386cb7f3",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1849
      },
      "writtenOn": "2025-01-07T12:42:18Z",
      "side": 1,
      "message": "I ran the test without applying https://gerrit.fd.io/r/c/vpp/+/41721/3 and the test passed. Was it expected?",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "b69e25a5_0da0b954",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1854
      },
      "writtenOn": "2025-01-07T12:51:10Z",
      "side": 1,
      "message": "No it is not expected - in my case the test was failing as nothing was captured and in the log from the test you could see that there was no match in the policy. Let me double check yet.",
      "parentUuid": "33d1117c_386cb7f3",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "24ac01c0_9fca7c09",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1854
      },
      "writtenOn": "2025-01-07T14:15:45Z",
      "side": 1,
      "message": "I\u0027ve double checked. For me without ip6 matching patch test is failing:\n  1 ERROR: test_ipsec6_spd_inbound_tun_protect (test_ipsec_spd_fp_input.IPSec6SpdTestCaseTunProtect)\n  2 ------------------------------------------------------------------------------\n  3 Traceback (most recent call last):\n  4   File \"/opt/pibr/github/networking.dataplane.fdio.vpp/test/test_ipsec_spd_fp_input.py\", line 924, in test_ipsec6_spd_inbound_tun_protect\n  5     recv_pkts \u003d self.send_and_expect(self.tun_if, send_pkts, self.pg1)\n  6   File \"/opt/pibr/github/networking.dataplane.fdio.vpp/test/framework.py\", line 558, in send_and_expect\n  7     rx \u003d output.get_capture(n_rx, filter_out_fn\u003dfilter_out_fn)\n  8   File \"/opt/pibr/github/networking.dataplane.fdio.vpp/test/vpp_pg_interface.py\", line 350, in get_capture\n  9     raise Exception(f\"No packets captured on {name} (timeout \u003d {timeout}s)\")\n 10 Exception: No packets captured on pg1 (timeout \u003d 1s)\n 11\n 12 14:13:27,865 Symlink to failed testcase directory: /tmp/vpp-failed-unittests/vpp-unittest-IPSec6SpdTestCaseTunProtect-FAILED -\u003e vpp-unittest-IPSec6SpdTestCaseTunProtect\n\nWhich is expeceted behavior",
      "parentUuid": "b69e25a5_0da0b954",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "84f760f5_2e3efc23",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1849
      },
      "writtenOn": "2025-01-08T08:53:45Z",
      "side": 1,
      "message": "Can you send me the command for your test?",
      "parentUuid": "24ac01c0_9fca7c09",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "3f706dca_60a775fd",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1854
      },
      "writtenOn": "2025-01-08T09:04:29Z",
      "side": 1,
      "message": "make TEST\u003dtest_ipsec_spd_fp_input.IPSec6SpdTestCaseTunProtect test",
      "parentUuid": "84f760f5_2e3efc23",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "373fec67_72167b45",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1849
      },
      "writtenOn": "2025-01-08T11:13:13Z",
      "side": 1,
      "message": "Thanks. I used the same command (test-debug) and there was a compiling error in xdp-tools \"error: unknown warning option \u0027-Wno-maybe-uninitialized\u0027; did you mean \u0027-Wno-uninitialized\u0027? [-Werror,-Wunknown-warning-option]\" so I changed to \"-Wno-uninitialized\" and the compile is now ok. With that the test passed. BTW what is your clang version?",
      "parentUuid": "3f706dca_60a775fd",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "7cb11d21_ffcab21b",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1854
      },
      "writtenOn": "2025-01-08T11:37:13Z",
      "side": 1,
      "message": "My clang is Ubuntu clang version 14.0.0-1ubuntu1.1.\nBut the behavior you described is very strange, do you have any log from this test?There should be policy matching and in order to have this test passing there must be a match to a policy - I would like to see the trace for encrypted packets - in my case without patch applied it looks as follows:\n\n00:00:00:478656: pg-input\n  stream 0, 218 bytes, sw_if_index 1, next_node ethernet-input\n  current data 0, length 218, buffer-pool 0, ref-count 1, trace handle 0x0\n  00000000: 02fe62550fac02010000ff0286dd6000000000a43240fd010001000000000000\n  00000020: 000000000002fd01000100000000000000000000000100000bb9000000016759\n  00000040: d8883349d09510026b6d4811a73e1822a1b4fc4c3ea1971b80b48528db04ee06\n  00000060: e348b97f46c0f9873b07c0412093454bdeb8bb0e27cbff4c814e4b1fffba9d35\n00:00:00:478660: ethernet-input\n  frame: flags 0x1, hw-if-index 1, sw-if-index 1\n  IP6: 02:01:00:00:ff:02 -\u003e 02:fe:62:55:0f:ac\n00:00:00:478662: ip6-input\n  IPSEC_ESP: fd01:1::2 -\u003e fd01:1::1\n    tos 0x00, flow label 0x0, hop limit 64, payload length 164\n00:00:00:478663: ipsec6-input-feature\n  IPSEC_ESP: sa_id 0 type: 0 spd 1 policy -1 spi 3001 (0x00000bb9) seq 1\n00:00:00:478665: error-drop\n  rx:pg0\n00:00:00:478666: drop\n  null-node: blackholed packets",
      "parentUuid": "373fec67_72167b45",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "9b10f837_18672b15",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 1
      },
      "lineNbr": 0,
      "author": {
        "id": 1854
      },
      "writtenOn": "2025-01-09T09:45:58Z",
      "side": 1,
      "message": "Done",
      "parentUuid": "7cb11d21_ffcab21b",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "fb82c087_0e1bd0e0",
        "filename": "test/test_ipsec_spd_fp_input.py",
        "patchSetId": 1
      },
      "lineNbr": 894,
      "author": {
        "id": 1849
      },
      "writtenOn": "2025-01-07T12:42:18Z",
      "side": 1,
      "message": "The line behind \"(protect)...\" was displayed in the test output. Please either remove or comment with \"# ...\" Instead",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "376450df_93bd1223",
        "filename": "test/test_ipsec_spd_fp_input.py",
        "patchSetId": 1
      },
      "lineNbr": 894,
      "author": {
        "id": 1854
      },
      "writtenOn": "2025-01-08T08:34:31Z",
      "side": 1,
      "message": "Done",
      "parentUuid": "fb82c087_0e1bd0e0",
      "revId": "3312893936332af034f061acbb2b38d67256b5e1",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2"
    }
  ]
}